src/argilla/client/feedback/dataset/remote/filtered.py

```diff
@@ -92,4 +94,24 @@ def __init__(
         )
 
     def delete(self) -> None:
-        raise NotImplementedError("`delete` does not work for filtered datasets.")
+        raise NotImplementedError(
+            "`delete` does not work for filtered datasets. First call `filter_reset()` and then `delete()`."
+        )
+
+    def filter_by(self, *args, **kwargs) -> "FilteredRemoteFeedbackDataset":
+        warnings.warn(message="Removing old filters and applying new ones.", stacklevel=2)
+        super.filter_by(*args, **kwargs)
+
+    def filter_reset(self) -> "FilteredRemoteFeedbackDataset":
+        super().__init__(
+            client=self._client,
+            id=self.id,
+            name=self.name,
+            workspace=self.workspace,
+            created_at=self.created_at,
+            updated_at=self.updated_at,
+            fields=self.fields,
+            questions=self.questions,
+            guidelines=self.guidelines,
+            filters=[],
+        )
```

Comment:
Hmmm this seems a bit unclear to me, as ideally users should `filtered_dataset = remote_dataset.filter_by(...)` so there should be no need on adding a method to go back to `RemoteFeedbackDataset` IMO

Reply:
Yes, I also wasn't sure about this but normally when data wrangling, these kinds of operators are nice, however the duplicated `filter_by` I added, should do the job nicely, I agree with removing this too. Was just playing around.